managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1705,7 +1710,8 @@ public void asyncClose(final AsyncCallbacks.CloseCallback callback, final Object
 
         lock.readLock().lock();
         try {
-            if (!individualDeletedMessages.isEmpty()) {
+            if (cursorLedger != null && protobufFormat == ZNodeProtobufFormat.Text
```

Comment:
If you're writing in text format, we don't put the current position + individually deleted message ranges into the z-node, but fallback to append to current ledger, where the info is stored in binary format